src/LogicItLab/Salesforce/MapperBundle/WsdlValidator.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace LogicItLab\Salesforce\MapperBundle;
+
+use LogicItLab\Salesforce\MapperBundle\Annotation\AnnotationReader;
+use Symfony\Component\Finder\Finder;
+
+class WsdlValidator
+{
+    /** @var AnnotationReader */
+    private $annotationReader;
+
+    /** @var string */
+    private $baseProjectDir;
+
+    /** @var string */
+    private $modelDirPath;
+
+    /** @var string */
+    private $wsdlPath;
+
+    /** @var string */
+    private $namespacePrefix;
+
+    /** @var string */
+    private $wsdlContents;
+
+    /**
+     * @param AnnotationReader $annotationReader
+     * @param string $baseProjectDir
+     * @param string $modelDirPath
+     * @param string $wsdlPath
+     * @codeCoverageIgnore
+     */
+    public function __construct(AnnotationReader $annotationReader, string $baseProjectDir, string $modelDirPath, string $wsdlPath, string $namespacePrefix = "")
+    {
+        $this->annotationReader = $annotationReader;
+        $this->baseProjectDir = $baseProjectDir;
+        $this->modelDirPath = $modelDirPath;
+        $this->wsdlPath = $wsdlPath;
+        $this->namespacePrefix = $namespacePrefix;
+    }
+
+    public function validate(): array
+    {
+        $missingFields = [];
+
+        foreach ($this->getAllClassAnnotations() as $annotation) {
+            $objectName = $annotation['object']->name;
+            $fieldNames = $this->getFieldNames($annotation['fields']);
+
+            if (!$objectName || !$fieldNames) {
+                continue;
+            }
+
+            foreach ($fieldNames as $fieldName) {
+                if ($this->hasField($objectName, $fieldName) === false) {
+                    $missingFields[$objectName][] = $fieldName;
+                }
+            }
+        }
+
+        return $missingFields;
+    }
+
+    private function getObject(string $objectName): ?string
+    {
+        preg_match("/<complexType name=\"$objectName\">([\w\W]*)<\/complexType>/", $this->getWsdlContents(), $matches);
+
+        return $matches[0] ?? null;
+    }
+
+    private function hasField(string $objectName, string $fieldName): bool
+    {
+        return strpos($this->getObject($objectName), sprintf('<element name="%s"', $fieldName)) !== false;
+    }
+
+    private function getWsdlContents(): string
+    {
+        if (!$this->wsdlContents) {
+            $this->wsdlContents = file_get_contents($this->wsdlPath);
+        }
+
+        return $this->wsdlContents;
+    }
+
+    private function getAllClassAnnotations(): array
+    {
+        return array_map(function ($className) {
+            return $this->annotationReader->getSalesforceProperties($className);
+        }, $this->getAllClassNames());
+    }
+
+    private function getAllClassNames(): array
+    {
+        $classNames = [];
+        $AllFiles = Finder::create()->files()->in($this->baseProjectDir . $this->modelDirPath)->name('*.php');
```

Comment:
I'd probably use something like `**/Document/*.php` to get only the classes we want. Not sure if the Finder supports `glob` function syntax though.